daffodil-tdml-processor/src/main/scala/org/apache/daffodil/tdml/processor/Runtime2TDMLDFDLProcessor.scala

```diff
@@ -113,6 +113,11 @@ final class Runtime2TDMLDFDLProcessorFactory private(
       // Compile the generated code into an executable
       val executable = generator.compileCode(codeDir)
 
+      // There is is no way to clean up TDMLDFDLProcessors, they are just garbage
+      // collected when no longer needed. So recursively mark all generated
+      // files as deleteOnExit so we at least clean them up when the JVM exits
+      os.walk.stream(tempDir).foreach { _.toIO.deleteOnExit() }
```

Comment:
Have you tested that this line actually cleans up all temporary files used by runtime2 tests?  If not, I'll still approve but I'll also check for left-over temporary files during my next work on runtime2 just in case.

In general, Daffodil tests aren't very good about removing temporary files.  I always see a lot of Daffodil-created temporary files in /tmp after running "sbt test" and/or "sbt IntegrationTest/test".

Reply:
> Have you tested that this line actually cleans up all temporary files used by runtime2 tests? If not, I'll still approve but I'll also check for left-over temporary files during my next work on runtime2 just in case.

Yep, I did actually test and verify that it works. Note that it's not enough to have delteOnExit() on just the root directory, it's needs to be on the directory and everything inside it.

> In general, Daffodil tests aren't very good about removing temporary files. I always see a lot of Daffodil-created temporary files in /tmp after running "sbt test" and/or "sbt IntegrationTest/test".

Yeah, we're really bad about this. I too have noticed how much we pollute /tmp. We really need to clean everything up, or at least put everything in /tmp/daffodil/ or something so everything is in one place.

Reply:
We need a JIRA ticket for this. 

Honestly, the exact right idiom for creating and cleaning up temp files is not obvious to me. E.g., the fact that you can't just call deleteOnExit is news. 

We should put a temp-files section in our coding-style-guide on wiki.